scripts/dts/extract/globals.py

```diff
@@ -350,6 +388,15 @@ def extract_controller(node_path, prop, prop_values, index,
             continue
 
         cell_parent = phandles[elem[0]]
+
+        if 'gpio-map' in reduced[cell_parent]['props']:
+            # parent is a 'gpio nexus node' (ie has gpio-map)
+            try:
+                # controller is to be found in the map, using elem[1] as index
+                cell_parent = gpio_map_lookup(cell_parent, elem[1])
+            except IndexError:
+                raise Exception("GPIO index not available in gpio-map")
```

Comment:
Maybe `gpio_map_lookup()` could return `None` or something instead if the controller isn't found. Reading this quickly, you wonder if the `IndexError` is there to guard the `elem[1]` lookup or the like.

Exceptions for control flow often gets confusing.

Reply:
So I reworked a bit, but I'm not finding that much better in terms of exception, but seems more coherent in the dt object type handling. Let me know.